util/dvsim/Scheduler.py

```diff
@@ -233,44 +104,372 @@ def on_sigint(signal_received, frame):
         finally:
             signal(SIGINT, old_handler)
 
-        # We got to the end without anything exploding. Return the results for our jobs.
+        # We got to the end without anything exploding. Return the results.
         return self.item_to_status
 
+    def _schedule_items(self, items):
+        '''Convert a heterogenous list of items to run into list of tasks
+        segregated by target and cfg (DUT).'''
 
-class Scheduler:
-    '''An object to run one or more Deploy items'''
+        # Scheduled items are ordered by targets.
+        scheduled = OrderedDict()
```

Comment:
I don't think there's any need for an `OrderedDict` here. Python 3.6 gives the ordering stability that I think you're using to plain `dict`s.